tools/mtouch/Target.cs

```diff
@@ -402,46 +406,50 @@ void ComputeListOfAssemblies (HashSet<string> assemblies, AssemblyDefinition ass
 			// Custom Attribute metadata can include references to other assemblies, e.g. [X (typeof (Y)], 
 			// but it is not reflected in AssemblyReferences :-( ref: #37611
 			// so we must scan every custom attribute to look for System.Type
-			GetCustomAttributeReferences (assembly, assemblies, exceptions);
-			GetCustomAttributeReferences (main, assemblies, exceptions);
+			GetCustomAttributeReferences (main, assembly, assemblies, exceptions);
+			GetCustomAttributeReferences (main, main, assemblies, exceptions);
 			if (main.HasTypes) {
 				foreach (var ca in main.GetCustomAttributes ())
-					GetCustomAttributeReferences (ca, assemblies, exceptions);
+					GetCustomAttributeReferences (main, ca, assemblies, exceptions);
 			}
 		}
 
-		void GetCustomAttributeReferences (ICustomAttributeProvider cap, HashSet<string> assemblies, List<Exception> exceptions)
+		void GetCustomAttributeReferences (ModuleDefinition main, ICustomAttributeProvider cap, HashSet<string> assemblies, List<Exception> exceptions)
 		{
 			if (!cap.HasCustomAttributes)
 				return;
 			foreach (var ca in cap.CustomAttributes)
-				GetCustomAttributeReferences (ca, assemblies, exceptions);
+				GetCustomAttributeReferences (main, ca, assemblies, exceptions);
 		}
 
-		void GetCustomAttributeReferences (CustomAttribute ca, HashSet<string> assemblies, List<Exception> exceptions)
+		void GetCustomAttributeReferences (ModuleDefinition main, CustomAttribute ca, HashSet<string> assemblies, List<Exception> exceptions)
 		{
 			if (ca.HasConstructorArguments) {
 				foreach (var arg in ca.ConstructorArguments)
-					GetCustomAttributeArgumentReference (arg, assemblies, exceptions);
+					GetCustomAttributeArgumentReference (main, arg, assemblies, exceptions);
 			}
 			if (ca.HasFields) {
 				foreach (var arg in ca.Fields)
-					GetCustomAttributeArgumentReference (arg.Argument, assemblies, exceptions);
+					GetCustomAttributeArgumentReference (main, arg.Argument, assemblies, exceptions);
 			}
 			if (ca.HasProperties) {
 				foreach (var arg in ca.Properties)
-					GetCustomAttributeArgumentReference (arg.Argument, assemblies, exceptions);
+					GetCustomAttributeArgumentReference (main, arg.Argument, assemblies, exceptions);
 			}
 		}
 
-		void GetCustomAttributeArgumentReference (CustomAttributeArgument arg, HashSet<string> assemblies, List<Exception> exceptions)
+		void GetCustomAttributeArgumentReference (ModuleDefinition main, CustomAttributeArgument arg, HashSet<string> assemblies, List<Exception> exceptions)
 		{
 			if (!arg.Type.Is ("System", "Type"))
 				return;
 			var ar = (arg.Value as TypeReference)?.Scope as AssemblyNameReference;
 			if (ar == null)
 				return;
 			var reference_assembly = ManifestResolver.Resolve (ar);
+			if (reference_assembly == null) {
+				ErrorHelper.Warning (137, "Cannot find '{0}', referenced by an attribute in '{1}'.", ar.FullName, main.Name);
```

Comment:
_referenced by an attribute_ is a bit vague
showing the custom attribute (`ca`) would be a bit better
showing the ca provider (`cap`) would be even more useful to pinpoint the problem